src/history/historyTracker.ts

```diff
@@ -501,10 +501,10 @@ export class HistoryTracker {
 
     if (this.currentHistoryStep.changes.length === 0) {
       this.currentHistoryStepIndex--;
-    }
 
-    if (this.currentHistoryStepIndex === 0) {
-      return undefined;
+      if (this.currentHistoryStepIndex === 0) {
+        return undefined;
+      }
```

Comment:
What's the reasoning behind putting this in the if statement?

Reply:
It's not completely necessary, I simply thought it looked odd to have the exact same `if` statement twice in the same block. Nesting this statement makes the relationship between the second and third `if` statements more obvious (to me). I did the same change in the new function.

Reply:
Ah I didn't see the if statement above.